### PR TITLE
RHBPMS-4994 - org.drools.core.WorkItemHandlerNotFoundException doesn'…

### PR DESCRIPTION
**Pull request by mswiderski**
…t generate an execution error

### CODE REVIEW

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins execute full downstream build

**Comment by mswiderski** [comment]
Jenkins execute full downstream build

**Comment by mswiderski** [comment]
Jenkins execute full downstream build

**Comment by mswiderski** [comment]
Jenkins execute full downstream build

**Comment by mswiderski** [comment]
test failure (indexing) not related to the change

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/ExecutionErrorHandlerInterceptor.java [hunk 1]
```diff
@@ -43,13 +43,19 @@ public final RequestContext execute( Executable executable, RequestContext ctx )
         if (counter == null) {
             counter = new AtomicInteger( 0 );
             invocationsCounter.set( counter );
+            if (executionErrorManager != null) {
```
Comment (tsurdilo) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/ExecutionErrorHandlerInterceptor.java]: Sorry this is confusing to me - check for executionerrormanager being null is only relevant here if counter is also null? Same below in the finally method.

Comment (mswiderski) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/ExecutionErrorHandlerInterceptor.java]: yes, as this should only be done once for given transaction. interceptor can be many times on the stack as it is used by ksession and task service so we don't want it to be called many times.

File: jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/wih/AsyncWorkItemHandlerCmdCallback.java [hunk 2]
```diff
@@ -124,15 +124,20 @@ public Boolean execute(Context context) {
...
-        }
-        return errorManager.getHandler();
+        
+        return ((ExecutionErrorManagerImpl) errorManager).createHandler();
+    }
+    
+    private void closeErrorHandler(RuntimeManager manager) {
+        ExecutionErrorManager errorManager = ((AbstractRuntimeManager) manager).getExecutionErrorManager();
+        
+        ((ExecutionErrorManagerImpl) errorManager).closeHandler();
```
[the remark this answers is not in the document]
Comment (mswiderski) [reply jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/wih/AsyncWorkItemHandlerCmdCallback.java]: for now I want to keep it internal as ExecutionErrorManager is in kie-api/internal that could be too early to move. But I made a note and will think about it if this is the best place to create and close handlers.

File: jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/wih/AsyncWorkItemHandlerCmdCallback.java [hunk 3]
```diff
@@ -124,15 +124,20 @@ public Boolean execute(Context context) {
           logger.error("Error when handling callback from executor", e);  
         } finally {
             manager.disposeRuntimeEngine(engine);
+            closeErrorHandler(manager);
         }
     }
     
     private ExecutionErrorHandler getExecutionErrorHandler(RuntimeManager manager) {
         ExecutionErrorManager errorManager = ((AbstractRuntimeManager) manager).getExecutionErrorManager();
-        if (errorManager == null) {
```
[the remark this answers is not in the document]
Comment (mswiderski) [reply jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/wih/AsyncWorkItemHandlerCmdCallback.java]: good catch, fixed

**Comment by mswiderski** [comment]
@tsurdilo some comments applied and some replied, thanks

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by tsurdilo** [comment]
@mswiderski thanks looks good +1

**Pull request merged**